elasticapm/metrics/base_metrics.py

```diff
@@ -56,25 +57,36 @@ def __init__(self, client, tags=None):
         self._collect_timer = None
         super(MetricsRegistry, self).__init__()
 
-    def register(self, class_path):
+    def register(self, metricset: Union[str, type]) -> "MetricSet":
         """
         Register a new metric set
-        :param class_path: a string with the import path of the metricset class
+
+        :param metricset: a string with the import path of the metricset class,
+            or a class object that can be used to instantiate the metricset.
+            If a class object is used, you can use the class object or
+            `metricset.__name__` to retrieve the metricset using `get_metricset`.
+        :return: the metricset instance
         """
-        if class_path in self._metricsets:
-            return
+        class_id = metricset if isinstance(metricset, str) else metricset.__name__
```

Comment:
using only `metricset.__name__` here could lead to double registrations here I think. We should use the full class path instead, i.e.: 

```suggestion
        class_id = metricset if isinstance(metricset, str) else f"{metricset.__module__}.{metricset.__name__}"